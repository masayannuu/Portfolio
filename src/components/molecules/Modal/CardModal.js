import React from 'react'
import PropTypes from 'prop-types'

import Modal from './Modal'
import ModalLink from '../../atoms/Link/ModalLink'
import ModalBody from './ModalBody'

class CardModal extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      showModal: false,
      title: props.title,
      imageUrl: props.imageUrl,
      description: props.description,
      teamData: props.teamData,
      informationTags: props.informationTags
    }
    this.handleOpenModal = this.handleOpenModal.bind(this)
    this.handleCloseModal = this.handleCloseModal.bind(this)
  }

  handleOpenModal () {
    this.setState({ showModal: true })
  }

  handleCloseModal () {
    this.setState({ showModal: false })
  }

  render () {
    const isTag = typeof this.state.informationTags !== 'undefined'
    return (
      <div className="tk-a-otf-gothic-bbb-pr6n">
        <ModalLink onClick={this.handleOpenModal} />
        <Modal
          handleCloseModal={this.handleCloseModal}
          showModal={this.state.showModal}
          title={this.state.title}
        >
          <ModalBody
            imageUrl={this.state.imageUrl}
            description={this.state.description}
            teamData={this.state.teamData}
            informationTags={this.state.informationTags}
            isTag={isTag}
          />
        </Modal>
      </div>
    )
  }
}

CardModal.propTypes = {
  title: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  teamData: PropTypes.shape({
    number: PropTypes.number,
    roll: PropTypes.string,
    work: PropTypes.string
  }).isRequired
}

export default CardModal
