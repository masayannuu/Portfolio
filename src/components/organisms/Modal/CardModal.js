import React from 'react'

import Modal from './Modal'
import ModalLink from '../../atoms/Link/ModalLink'

class CardModal extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      showModal: false,
      title: props.title
    }
    this.handleOpenModal = this.handleOpenModal.bind(this)
    this.handleCloseModal = this.handleCloseModal.bind(this)
  }

  handleOpenModal () {
    this.setState({ showModal: true })
  }

  handleCloseModal () {
    this.setState({ showModal: false })
  }

  render () {
    return (
      <div>
        <ModalLink onClick={this.handleOpenModal} />
        <Modal
          handleCloseModal={this.handleCloseModal}
          showModal={this.state.showModal}
          title={this.state.title}
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sit amet justo in arcu
            efficitur malesuada nec ut diam. Aenean a iaculis eros. Proin nec purus congue, rutrum
            sapien id, sodales ante. Nam imperdiet sapien pretium leo dapibus euismod. Ut ac
            venenatis nunc. Praesent viverra purus vel lacus ullamcorper porta a a augue. Proin
            rhoncus tempus leo sed ultricies. In luctus aliquam placerat. Cras efficitur enim vitae
            vulputate consequat. Nulla tellus est, fringilla quis nisi eu, aliquam finibus eros.
          </p>
        </Modal>
      </div>
    )
  }
}

export default CardModal
