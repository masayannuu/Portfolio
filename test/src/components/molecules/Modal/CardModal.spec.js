import React from 'react'
import Enzyme from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import CardModal from '../../../../../src/components/molecules/Modal/CardModal'

const description =
  'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字'

const team_data = {
  number: 3,
  roll: '設計・開発主担当',
  work:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ'
}

describe('Modal CardModal Molecules', () => {
  it('should render CardModal', () => {
    const wrapper = Enzyme.shallow(<CardModal
      title="File Transfer Application"
      imageUrl="https://bulma.io/images/placeholders/1280x960.png"
      description={description}
      teamData={team_data}
      />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
