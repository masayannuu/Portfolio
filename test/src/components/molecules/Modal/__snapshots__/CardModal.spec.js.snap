// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal CardModal Molecules should render CardModal 1`] = `
<div
  className="tk-a-otf-gothic-bbb-pr6n"
>
  <ModalLink
    onClick={[Function]}
  />
  <Modal
    handleCloseModal={[Function]}
    showModal={false}
    title="File Transfer Application"
  >
    <ModalBody
      description="この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字"
      imageUrl="https://bulma.io/images/placeholders/1280x960.png"
      isTag={false}
      teamData={
        Object {
          "number": 3,
          "roll": "設計・開発主担当",
          "work": "この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ",
        }
      }
    />
  </Modal>
</div>
`;
